tests/garage/tf/models/test_model.py

```diff
@@ -0,0 +1,178 @@
+import pickle
+
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.core.mlp import mlp
+from garage.tf.models.base import TfModel
+from tests.fixtures import TfGraphTestCase
+
+
+class SimpleModel(TfModel):
+    def __init__(self, output_dim=2, hidden_sizes=(4, 4), name=None):
+        super().__init__(name)
+        self._output_dim = output_dim
+        self._hidden_sizes = hidden_sizes
+
+    def network_output_spec(self):
+        return ['state', 'action']
+
+    def _build(self, obs_input):
+        state = mlp(obs_input, self._output_dim, self._hidden_sizes, 'state')
+        action = mlp(obs_input, self._output_dim, self._hidden_sizes, 'action')
+        return state, action
+
+
+class ComplicatedModel(TfModel):
+    def __init__(self, output_dim=2, name=None):
+        super().__init__(name)
+        self._output_dim = output_dim
+        self._simple_model_1 = SimpleModel(output_dim=4)
+        self._simple_model_2 = SimpleModel(
+            output_dim=output_dim, name='simple_model_2')
+
+    def network_output_spec(self):
+        return ['action']
+
+    def _build(self, obs_input):
+        h1, _ = self._simple_model_1.build(obs_input)
+        return self._simple_model_2.build(h1)[1]
+
+
+class TestModel(TfGraphTestCase):
+    def test_model_creation(self):
+        input_var = tf.placeholder(tf.float32, shape=(None, 5))
+        model = SimpleModel(output_dim=2)
+        outputs = model.build(input_var)
```

Comment:
where is `outputs` used?

Reply:
in the `assert`

Reply:
I still can’t see which assert. The two asserts here use out and model_out

Reply:
`output` are the output Tensors. It's in the `feed_dict`. I am trying to show the outputs of `model.build()` are actually same as `model.networks['default'].outputs`. Users can do either way to access it.

Reply:
I guess my question is, can we pass outputs as the fetches in sess.run?

Reply:
I c.